### PR TITLE
miner: compute tx fees for blocks

### PR DESCRIPTION
**Pull request by MariusVanDerWijden**
computes and returns the payload fees for a block.
Spec for it is here: https://github.com/ethereum/execution-apis/pull/314

### CODE REVIEW

File: core/beacon/types.go [hunk 1]
```diff
@@ -183,7 +185,26 @@ func ExecutableDataToBlock(params ExecutableDataV1) (*types.Block, error) {
 
 // BlockToExecutableData constructs the executableDataV1 structure by filling the
 // fields from the given block. It assumes the given block is post-merge block.
-func BlockToExecutableData(block *types.Block) *ExecutableDataV1 {
+func BlockToExecutableData(block *types.Block, fees *big.Int) *ExecutableDataV1 {
+	if fees != nil {
```
Comment (lightclient) [inline core/beacon/types.go:189]: Is it better to create the object first and then add fees if not nil?

**Comment by metachris** [comment]
fwiw, the [mev-boost block scoring](https://collective.flashbots.net/t/block-scoring-for-mev-boost-relays) is likely to standardize on payment transaction value, and not on balance diff over the whole block. 5 out of 7 relays already support that way of defining the bid value.

File: core/beacon/types.go [hunk 2]
```diff
@@ -183,7 +185,26 @@ func ExecutableDataToBlock(params ExecutableDataV1) (*types.Block, error) {
...
+			GasUsed:       block.GasUsed(),
+			BaseFeePerGas: block.BaseFee(),
+			Timestamp:     block.Time(),
+			ReceiptsRoot:  block.ReceiptHash(),
+			LogsBloom:     block.Bloom().Bytes(),
+			Transactions:  encodeTransactions(block.Transactions()),
+			Random:        block.MixDigest(),
+			ExtraData:     block.Extra(),
+			Fees:          fees,
+		}
```
Comment (dcposch) [inline core/beacon/types.go:206]: maybe a dumb question, but how is this block different from adding `Fees: fees,` below?

nil is the default value for `Fees`

Comment (MariusVanDerWijden) [reply core/beacon/types.go:206]: I know, I'm stupid, wrote this after coming home from bogota in the middle of the night^^

**Comment by holiman** [comment]
@MariusVanDerWijden is this still needed? If so, please rebase and un-draft, otherwise close

**Comment by MariusVanDerWijden** [comment]
Yes it is needed, the spec is not final so I am not going to un-draft it. But I'm rebasing it rn

**Comment by MariusVanDerWijden** [comment]
closing and reopening against lightclients withdrawal-timestamp branch

**Pull request closed without merging**